doc/design/executors.md

```diff
@@ -0,0 +1,187 @@
+# Executors
+
+Executors are responsible for coordinating actor execution, global garbage collection, and managing
+I/O operations. During runtime, a Hemlock process has an on-to mapping of executor to CPU core. The
+number of executors is a dynamic property of a Hemlock process. In cases where enough
+parallelization exists, the runtime creates enough executors to utilize all cores, making it a
+one-to-one mapping as well.
+
+## Lookup array
+
+Live executors are discoverable via a global array. The array is large enough to hold one pointer
+for each of the possible executors. The current number of executors `n` indicates that executors
+`[0,n)` in the array are live. Upon initialization, live exectors are atomically emplaced (via `m`)
+in the list at `executors[n]` while `n` is incremented.
+
+```c
+pthread_mutex_t mutex;
+size_t n;
+hm_executor_t executors[];
+
+void executors_emplace(hm_executor_t * executor) {
+   pthread_mutex_lock(&mutex)
+   executors[n] = executor;
+   n++;
+}
+```
+
```

Comment:
```suggestion
```
Let's just drop this code from the doc. We're not necessarily going to use a mutex here, and this deadlocks as written. :wink: